pkg/addon/source.go

```diff
@@ -392,23 +150,63 @@ func NewAsyncReader(baseURL, bucket, subPath, token string, rdType ReaderType) (
 			bucketEndPoint = fmt.Sprintf(bucketTmpl, ossURL.Scheme, bucket, ossURL.Host)
 		}
 		return &ossReader{
-			baseReader:     bReader,
 			bucketEndPoint: bucketEndPoint,
 			path:           subPath,
 			client:         resty.New(),
 		}, nil
 	}
-	return nil, errors.New("addon registry invalid")
+	return nil, fmt.Errorf("invalid addon registry type '%s'", rdType)
+}
+
+// BuildReader will build a AsyncReader from registry, AsyncReader are needed to read addon files
+func (r Registry) BuildReader() (AsyncReader, error) {
```

Comment:
```suggestion
func (r *Registry) BuildReader() (AsyncReader, error) {
```

We ususally use pointer object as the method owner to avoid object copy.

Reply:
The same below.

Reply:
done